src/main/java/frc/robot/utils/VisionUtils.java

```diff
@@ -13,6 +18,20 @@
  */
 public class VisionUtils {
 
+    /**
+     * Sets up a listener for the vision system
+     * 
+     * @param listener The listener to call when a new pose is received
+     */
+    public static void setupVisionListener(Consumer<Pair<Pose2d, Double>> listener) {
+        NetworkTableInstance.getDefault().addListener(VisionConstants.LIMELIGHT_TABLE.getEntry("botpose"),
+                EnumSet.of(Kind.kValueAll), (event) -> {
+                    Pair<Pose2d, Double> pose = getVisionPose();
+                    if (pose != null)
+                        listener.accept(pose);
+                });
+    }
```

Comment:
what does that have to do with chassis?

Reply:
We update the odometer according to the vision so I set up a listener to the limelight

Reply:
ok, next time please separate MR's.